airflow-core/src/airflow/api_fastapi/execution_api/routes/task_instances.py

```diff
@@ -308,6 +310,15 @@ def ti_update_state(
         updated_state = ti_patch_payload.state
         query = TI.duration_expression_update(ti_patch_payload.end_date, query, session.bind)
         query = query.values(state=updated_state)
+
+        if updated_state == TerminalTIState.FAILED:
+            ti = session.get(TI, ti_id_str)
+            ser_dag = request.app.state.dag_bag.get_dag(dag_id)
+            if ser_dag and getattr(ser_dag, "fail_fast", False):
+                task_dict = getattr(ser_dag, "task_dict")
```

Comment:
This should work, but okay if we don't want to change it now
```suggestion
                task_dict =ser_dag.task_dict
```

Reply:
Same here, will handle tomorrow